fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/synthetic/SyntheticRecordPlannerTest.java

```diff
@@ -1229,7 +1403,7 @@ void updateSyntheticIndexesWhenDisabled() throws Exception {
     }
 
     @Test
-    void updateSyntheticIndexesWhenWriteOnly() throws Exception {
+    void updateRecordWhenSyntheticIndexesIsWriteOnly() throws Exception {
```

Comment:
These last two tests had very similar names. I've renamed them to make their difference more apparent (namely that one is about record updates and the other about record inserts). Other than adding assertions about the number of synthetic type plans, the tests are unchanged.